common/src/Model/Polyhedron_ConvexHull.h

```diff
@@ -918,14 +942,25 @@ namespace TrenchBroom {
                 boundary.push_back(h3);
                 edge->setSecondEdge(h2);
 
+                const vm::plane<T,3> plane = makePlaneFromBoundary(boundary);
                 if (it != std::end(seam)) {
-                    const auto [valid, plane] = vm::from_points(top->position(), v2->position(), v1->position());
-                    assert(valid); unused(valid);
-
                     auto* next = *it;
 
+                    const auto checkEdge = [&](const auto* e) {
+                        const auto& point = e->firstVertex()->position();
+                        
+                        // Is the point to be added inside the plane of the face being created?
+                        if (plane.point_status(point) != vm::plane_status::inside) {
+                            return false;
+                        }
+                        
+                        // Is the point to be added colinear with the first two vertices of the face being created?
+                        const bool valid = std::get<0>(vm::from_points(top->position(), point, v1->position()));
```

Comment:
This line is unreadable, if the function were renamed to `vm::plane_from_points` that would fix the problem (mostly, `const auto [valid, plane] = vm::plane_from_points(top->position(), point, v1->position())` would be better).

Reply:
Changed in 4336c58d86f0cdd24d8964db4617f657c375b72a.